packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2061,55 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (this.match(tt._with)) {
+      this.expectPlugin("moduleAttributes");
+      this.next();
+    } else {
+      if (this.hasPlugin("moduleAttributes")) return [];
+      return null;
+    }
+    const attrs = [];
+    const attributeMap = Object.create(null);
```

Comment:
I think it suffices to model `attributeMap` as a `Set<string>`. The value of `attributeMap` is redundant.

Reply:
Set is great. Thank you 👍 Have replaces `Object.create` with `Set` now.